### PR TITLE
Turret Presets

### LINKED ISSUE #159: Set desired pitch angle to zero when disabled

### PR DESCRIPTION
**Pull request by alicekuznetsov**
Closes #159

### CODE REVIEW

**Comment by alicekuznetsov** [comment]
@FRCTeam3255/code-reviewers Currently thinking about our trap implementation. Although I think its nice that we have logic to "shoot into the trap," I find it very unlikely that we'll actually be doing that. It also makes the code simpler if we don't. 

I'd like to propose making all of the logic simpler by turning the trap score into a preset (like subwoofer & the amp override). For reference, it's currently similar to "Lock Amp"

Another similar topic of discussion: the AMP!!!
Our current desired implementation of the Amp in my brain (following our controller layout, not the code) is as follows:
- "Lock Amp" = Shooter and pitch are constant values. The only auto-aim is the turret
- "Amp Preset" = Shooter, pitch, and turret are constant values

At this point, what is the Pro of having the lock amp button? I think it won't get used, just adding complexity to the driving experience (which we really want to minimize with our current situation). I propose that we remove the "Lock Amp" button entirely (at least for now) and remap the controller accordingly. 

also I am taking all of my misc. clean-up changes and moving them to their own pr because i realised i made so many that i just want in main at this point so look out for that pr in like an hour k thx (#162)

**Comment by alicekuznetsov** [comment]
Testing if LockPitch gives us good angles in SIM:
### WING
Observed Value: 18 degrees
Statistic: 19.1 degrees
image

### CENTERLINE
Observed Value: 16 degrees
Statistic: 13.96
image

Really bad desmos graph of those here: https://www.desmos.com/calculator/56mrsfygsj

**Comment by TaylerUva** [comment]
@ACat701 I agree. I think we just have lock speaker (pose yaw and pitch and static rpm). And then AMP preset (static pitch, yaw, rpm), Trap preset (static pitch, yaw, rpm) and sub preset (static pitch, yaw, rpm). I don't think the robot is physically capable of doing AMP, Trap or Sub of those with anything but static presets. Only speaker can be dynamic.

**Comment by alicekuznetsov** [comment]
Changes have been implemented 👍

File: src/main/java/frc/robot/RobotContainer.java [hunk 1]
```diff
@@ -77,10 +78,11 @@ public RobotContainer() {
         .setDefaultCommand(new Drive(subDrivetrain, conDriver.axis_LeftY, conDriver.axis_LeftX, conDriver.axis_RightX,
             isPracticeBot()));
 
-    subTurret.setDefaultCommand(new LockTurret(subTurret, subDrivetrain));
+    // subTurret.setDefaultCommand(new LockTurret(subTurret, subDrivetrain));
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotContainer.java]: Is this desired or just for testing?

File: src/main/java/frc/robot/RobotContainer.java [hunk 2]
```diff
@@ -106,25 +109,53 @@ private void configureDriverBindings(SN_XboxController controller) {
...
+    // Trap Preset
+    controller.btn_Y.onTrue(Commands.runOnce(() -> setLockedLocation(LockedLocation.NONE)).alongWith(
+        Commands.runOnce(() -> subShooter.setDesiredVelocities(prefShooter.leftShooterTrapVelocity.getValue(),
+            prefShooter.rightShooterTrapVelocity.getValue())))
+        .alongWith(
+            Commands.runOnce(() -> subTurret.setTurretAngle(prefTurret.turretTrapPresetPos.getValue())))
+        .alongWith(Commands.runOnce(() -> subPitch.setPitchAngle(prefPitch.pitchTrapAngle.getValue()))));
+
+    // TODO: ADD TO CONTROLLER MAP GRAPHIC. Is this where we want this control?
     controller.btn_LeftStick.whileTrue(new IntakeFromShooter(subShooter, subTransfer));
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotContainer.java]: This probably should be mapped else where now that we freed up buttons

File: src/main/java/frc/robot/RobotPreferences.java [hunk 3]
```diff
@@ -203,22 +204,36 @@ public static final class prefPitch {
...
+    /**
+     * <b>Units:</b> Degrees
+     */
+    public static final SN_DoublePreference pitchAmpAngle = new SN_DoublePreference("pitchAmpAngle", 52);
+
+    public static final SN_DoublePreference pitchWingAngle = new SN_DoublePreference("pitchWingAngle", 18);
+    public static final SN_DoublePreference pitchTrapAngle = new SN_DoublePreference("pitchTrapAngle",
+        18);
+
+    public static final SN_DoublePreference pitchCenterAngle = new SN_DoublePreference("pitchCenterAngle", 16);
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotPreferences.java:224]: What is this one used for?

File: src/main/java/frc/robot/subsystems/Shooter.java [hunk 4]
```diff
@@ -96,31 +106,54 @@ public double getRightShooterVelocity() {
...
+        && isRightShooterUpToSpeed();
+  }
+
+  public void setLeftDesiredVelocity(double desiredVelocity) {
+    desiredLeftVelocity = desiredVelocity;
+  }
+
+  public void setRightDesiredVelocity(double desiredVelocity) {
+    desiredRightVelocity = desiredVelocity;
+  }
```
Comment (TaylerUva) [inline src/main/java/frc/robot/subsystems/Shooter.java:138]: Any reason to have these individually?

**Comment by alicekuznetsov** [comment]
I changed the height of the speaker to make that test work. It probably did need to be shifted slightly, but not as much as I changed it. In hindsight, the real issue was this: (the todo comment cracks me up lmao)
```java
 // TODO: Update with real values (MUST DO BEFORE TESTING!)
    /**
     * The position, in meters, of the center of rotation for the pitch motor
     * relative to the center of the robot (Robot Coordinates).
     * 
     * @see <a href=
     *      "https://docs.wpilib.org/en/stable/docs/software/basic-programming/coordinate-system.html">Robot
     *      Coordinate System</a>
     */
    public static final Transform3d ROBOT_TO_PITCH = new Transform3d(
        new Translation3d(0, 0, 0),
        new Rotation3d(0, 0, 0));
```

@TaylerUva  Could you get me the values for this transform 3d (from the cad)? We'll ignore the whole "its on a turret" issue for now. Just get the position based on when the turret is pointing straight ahead

**Review by TaylerUva — DISMISSED** [review]

File: src/main/java/frc/robot/FieldConstants.java [hunk 5]
```diff
@@ -37,7 +37,7 @@ private static final class blueConstants {
     /**
      * The coordinate of the center of the blue speaker, in meters
      */
-    private static final Pose3d SPEAKER = new Pose3d(0, 5.547, 2.0525, new Rotation3d(0, 0, 0));
+    private static final Pose3d SPEAKER = new Pose3d(0, 5.547, 1.5525, new Rotation3d(0, 0, 0));
```
Comment (TaylerUva) [inline src/main/java/frc/robot/FieldConstants.java]: Update

File: src/main/java/frc/robot/FieldConstants.java [hunk 6]
```diff
@@ -51,7 +51,7 @@ private static final class redConstants {
     /**
      * The coordinate of the center of the red speaker, in meters
      */
-    private static final Pose3d SPEAKER = new Pose3d(16.5410515, 5.547, 2.0525, new Rotation3d(0, 0, 0));
+    private static final Pose3d SPEAKER = new Pose3d(16.5410515, 5.547, 1.5525, new Rotation3d(0, 0, 0));
```
Comment (TaylerUva) [inline src/main/java/frc/robot/FieldConstants.java]: Update

File: src/main/java/frc/robot/RobotContainer.java [hunk 7]
```diff
@@ -181,6 +212,15 @@ public static void logPDHValues() {
     }
   }
 
+  /**
+   * Sets all applicable subsystem's last desired location to their current
+   * location
+   */
+  public void clearSubsystemMovements() {
+    Commands.runOnce(() -> subPitch.setPitchAngle(subPitch.getPitchAngle()));
+    Commands.runOnce(() -> subTurret.setTurretAngle(subTurret.getAngle()));
```
Comment (TaylerUva) [inline src/main/java/frc/robot/RobotContainer.java]:
```suggestion
    subPitch.setPitchAngle(subPitch.getPitchAngle());
    subTurret.setTurretAngle(subTurret.getAngle());
```

**Review by TaylerUva — DISMISSED** [review]

**Review by TaylerUva — DISMISSED** [review]

**Review by TaylerUva — APPROVED** [review]

**Pull request merged**